src/crawlee/playwright_crawler/_playwright_crawler.py

```diff
@@ -149,6 +149,8 @@ async def _navigate(
         """
         async with context.page:
             # Navigate to the URL and get response.
+            if context.request.headers:
+                await context.page.set_extra_http_headers(context.request.headers.model_dump())
```

Comment:
The comment belongs to the next line.

```suggestion
            if context.request.headers:
                await context.page.set_extra_http_headers(context.request.headers.model_dump())
            # Navigate to the URL and get response.
```